src/test/compile-fail/imports/shadowing.rs

```diff
@@ -0,0 +1,35 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![feature(rustc_attrs)]
+#![allow(unused)]
+#![feature(item_like_imports)]
+
+mod foo { pub fn f() {} }
+mod bar { pub fn f() {} }
+
+pub fn f() -> bool { true }
+
+fn g() {
+    use foo::*;
+    use bar::*;
+    use f; // This shadows the globs
+    let _: bool = f();
+}
+
+fn h() {
+    use foo::*;
+    use bar::*;
+    fn f() -> bool { true } // This also shadows the globs
+    let _: bool = f();
+}
+
+#[rustc_error]
+fn main() {} //~ ERROR compilation successful
```

Comment:
Could you add to the run-pass situations where there appears to be shadowing but isn't because of namespaces, and where there is shadowing in one namespace, but not another.